go/cbor.go

```diff
@@ -804,7 +816,7 @@ func (dec *Decoder) decodeArray(rv DecodeValue, cborInfo byte, aux uint64) error
 		makeLength = int(aux)
 	}
 
-	dva, err = rv.CreateArray(makeLength)
+	dva, err = rv.CreateArray(int(min(uint64(makeLength), uint64(arrayBatch))))
```

Comment:
we might want to check here, if the expected length gets cut, we should error.

Reply:
This is only pre-allocation size, it's fine to have more entries here

Reply:
Yes, it will expand. LGTM.